pkg/profiler/profiler.go

```diff
@@ -132,6 +132,9 @@ func (p *CgroupProfiler) Stop() {
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 	level.Debug(p.logger).Log("msg", "stopping cgroup profiler")
+	if !prometheus.Unregister(p.missingStacks) {
```

Comment:
I just realized after merge, this is not exactly correct.
This unregisters from the default registry. It suppose to be register that registered the metric so the one in the profiler.